app/policies/user_context.rb

```diff
@@ -21,4 +21,20 @@ def group_uuids
       Insights::API::Common::RBAC::Service.paginate(api, :list_groups, :scope => 'principal').collect(&:uuid)
     end
   end
+
+  def self.current_user_context
+    Thread.current[:user_context]
+  end
+
+  def self.with_user_context(user_context)
+    saved_user_context   = Thread.current[:user_context]
+    self.current_user_context = user_context
+    yield
+  ensure
+    Thread.current[:user_context] = saved_user_context
+  end
+
+  def self.current_user_context=(user_context)
```

Comment:
Suggest moving the setter/getter methods next to each other.

Reply:
Done